gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -84,7 +87,7 @@ static GoogleHadoopFSInputStream create(
     return new GoogleHadoopFSInputStream(ghfs, fileInfo.getPath(), channel, statistics);
   }
 
-  private GoogleHadoopFSInputStream(
```

Comment:
instead of making this public, why not use the create method and mock GoogleHadoopFileSystem?

Reply:
There were total of three options 

1. mocking the transport layer.
2. mocking gcsio layer i.e GoogleCloudStorageFileSystem layer
3. mocking the constructor of GHFSInputStream and provide the mocked channel ( implemented in here).

Option 1 will require separate mocking for http and grpc layers. Given it's too much work and as unit which we are testing is just the exception thrown in channel. It not ideal to go via this route.

Option 2, just mocking GCSFileSystem will not be enough we have to also mock the underneath channel returned form GCSFileSystem layer which is an extra effort, but still doable.

Option 3, mocking the channel passed in GHFSInputStream. It is straight forward and seems the best option as for writing unit test in gcs layer we should be treating gcsio layer as blackbox. In doing so we have to expose the constructor of `GoogleHadoopFSInputStream`. It doesn't seem too much concerning to me as the only place we create the GHFSInputStream is in GHFS.create call.

Having said that if this is a concern we can move to Option 2.

Reply:
Updated to Option 2